### DISCUSSION TITLE
Where does winget get this ID name information?

### DISCUSSION THREAD

**Discussion by maisondasilva** [microsoft/winget-cli#3981] [Q&A]
I installed the beta version of WinRAR and noticed that "WinRAR archiver" appears, could you tell me where Winget gets this information so I can pass it on to the WinRAR developer to remove it and just keep "WinRAR"?

![image](https://github.com/microsoft/winget-cli/assets/11423362/10de3047-6103-4155-8119-cb50817c7d8e)

**Comment by maisondasilva** [comment]
@denelon Can you help me with this doubt?

Thanks

**Reply by Trenly** [reply]
That is the Product Code the application writes its Control Panel information underneath. When WinGet can't map an install package to one available from a source, the Product Code is used as the ID instead

**Reply by maisondasilva** [reply]
Control Panel not show "WinRAR archiver"

![image](https://github.com/microsoft/winget-cli/assets/11423362/03244a7c-6af1-4f47-a433-5e027f606234)

**Reply by Trenly** [reply]
Control Panel doesn’t show the product code.

All of the data you see in the control panel is written in the registry. One of the locations is `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\`. If you look at the registry key, there should be a sub-key (i.e folder) that is the product code - `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\WinRAR archiver`. Inside of that key are the `DisplayVersion`, `DisplayName`, and `Publisher` keys that are what you see in Control panel

**Reply by maisondasilva** [reply]
Yes, I had noticed this folder and sent it to him, but I thought that Winget gets it from another location!

![image](https://github.com/microsoft/winget-cli/assets/11423362/39b86a79-2d78-4af4-af9b-c9cd9ba69581)

**Comment by maisondasilva** [comment]
@Trenly It would not be correct for Winget to read DisplayName instead of parent key instead, namely the last part of:

`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\WinRAR archiver`

**Reply by Trenly** [reply]
The DisplayName is used as the Name of the application. Multiple applications could have the same display name. Product Codes (the name of the parent key) must be unique to an application. ID's also must be unique. This is why, when there is no mapping, the product code is used for the Id.

**Reply by maisondasilva** [reply]
Got it, thanks for the information!

**Reply by denelon** [reply]
I've been out for a few days, but it looks like @Trenly answered your questions @maisondasilva
